From 7586ac0e78eadec10ac0f205452a96f2cc1b8cb6 Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Tue, 10 Sep 2024 13:33:12 +0300
Subject: [PATCH] [FSW-9509] script: modified to take proper name

Currently script is parsing only macros with name mgmt.
Now we have mgmt for IO and compute die. So this change
will parse macros with mgmt and mmt_compute and create the bin
for that.

Change-Id: I1f6a1702d64f956fef3e334d5acb61bf6a0ae508
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-image-utility.sh b/board/habanalabs/fs1_arc/fs1-image-utility.sh
index 09c8a91..f0f2c31 100755
--- a/board/habanalabs/fs1_arc/fs1-image-utility.sh
+++ b/board/habanalabs/fs1_arc/fs1-image-utility.sh
@@ -96,13 +96,17 @@
 
 function hl_extract_zephyr_bins_sections {
 	local bins_file=$1
-	local app=$(basename ${bins_file} | cut -d. -f1 | cut -d_ -f2)
+	local app=$(basename ${bins_file} | cut -d. -f1 | cut -d_ -f2,3)
 	local sections=()
 	local bin_start
 	local bin_end
 	local sec_start
 	local sec_end
 
+	if ! echo "$app" | grep -q -E "io" && ! echo "$app" | grep -q -E "compute"; then
+		app=$(basename ${bins_file} | cut -d. -f1 | cut -d_ -f2)
+	fi
+
 	echo "Extracting sections for ${app} app"
 	while IFS=',' read -r bin_name bin_addr bin_size bin_align; do
 		echo -e "\tBinary ${bin_name} ($((${bin_size} * 1024))@${bin_addr})"
